lib/src/test/java/com/auth0/jwt/impl/JsonNodeClaimTest.java

```diff
@@ -207,6 +205,36 @@ public void shouldThrowIfListClassMismatch() throws Exception {
     }
 
     @Test
+    public void shouldGetMapValue() throws Exception {
+        Map<String, Object> map = new HashMap<>();
+        map.put("text", "extraValue");
+        map.put("number", 12);
+        map.put("boolean", true);
+        map.put("object", Collections.singletonMap("something", "else"));
+
+        JsonNode value = mapper.valueToTree(map);
+        Claim claim = claimFromNode(value);
+
+        assertThat(claim, is(notNullValue()));
+        Map<String, Object> backMap = claim.asMap();
+        assertThat(backMap, is(notNullValue()));
+        assertThat(backMap, hasEntry("text", (Object) "extraValue"));
+        assertThat(backMap, hasEntry("number", (Object) 12));
+        assertThat(backMap, hasEntry("boolean", (Object) true));
+        assertThat(backMap, hasKey("object"));
+        assertThat((Map<String, Object>) backMap.get("object"), IsMapContaining.hasEntry("something", (Object) "else"));
```

Comment:
I think I added the tests you're asking for, considering this PR adds only the ability to "get as map" and not the verify/create.